src/lib/storage/frame_of_reference/frame_of_reference_encoder.hpp

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <array>
+#include <limits>
+#include <memory>
+
+#include "storage/base_column_encoder.hpp"
+
+#include "storage/frame_of_reference_column.hpp"
+#include "storage/value_column.hpp"
+#include "storage/value_column/value_column_iterable.hpp"
+#include "storage/vector_compression/vector_compression.hpp"
+#include "types.hpp"
+#include "utils/enum_constant.hpp"
+
+namespace opossum {
+
+class FrameOfReferenceEncoder : public ColumnEncoder<FrameOfReferenceEncoder> {
+ public:
+  static constexpr auto _encoding_type = enum_c<EncodingType, EncodingType::FrameOfReference>;
+  static constexpr auto _uses_vector_compression = true;
```

Comment:
If I remember correctly, you don‘t *have* to compress your vectors, correct? If so, maybe `supports vector compression` would be better?

Reply:
This flag sets whether the encoding scheme uses a compressed vector as a back-end. The vector is always compressed, otherwise the memory footprint wouldn’t be reduced. So I think supports would suggest something different.

Reply:
What you wrote here should be a comment for that variable.

Reply:
@mrks is right, this would be another good place to point out that only vector compression creates any benefit from FOR.

Reply:
This looks still open to me.

Reply:
okay